src/libsyntax_ext/format.rs

```diff
@@ -901,25 +901,23 @@ pub fn expand_preparsed_format_args(ecx: &mut ExtCtxt<'_>,
 
     if !parser.errors.is_empty() {
         let err = parser.errors.remove(0);
-        let sp = fmt.span.from_inner_byte_pos(err.start.unwrap(), err.end.unwrap());
+        let sp = fmt.span.from_inner(err.span);
         let mut e = ecx.struct_span_err(sp, &format!("invalid format string: {}",
                                                      err.description));
         e.span_label(sp, err.label + " in format string");
         if let Some(note) = err.note {
             e.note(&note);
         }
-        if let Some((label, start, end)) = err.secondary_label {
-            let sp = fmt.span.from_inner_byte_pos(start.unwrap(), end.unwrap());
+        if let Some((label, span)) = err.secondary_label {
+            let sp = fmt.span.from_inner(span);
```

Comment:
I'm not sure whether this makes sense. `span` should already be based of the codemap, right? If not, then I'm not sure we should be using `Span`s there to avoid mixing `Span`s that are valid in some contexts and not others.

Reply:
It looks like the machinery in libfmt_macros is used from two places in the compiler: parsing `format_args!` literals in println, format, etc. and in parsing the attribute to `rustc_on_unimplemented`. The first has the relevant span information AFAICT fully, but the latter we don't appear to have good spans for it looks like we just have the span for all of `rustc_on_unimplemented = "..."` not just the `"..."` bit.

I'm happy to revert the bits which move libfmt_macros to using "fake" spans and back to `(SpanIndex, SpanIndex)`. I don't feel great about that since it seems worse than the inner spans.

Reply:
Could we new type wrap them? That way we won't be able to accidentally pass an `InnerSpan` somewhere a `Span` is expected. At the very least we should add generous comments about what's happening here. You're right that it seems better than the status quo.

Reply:
I'll do that. I agree that just `Span` is probably not great. I'll also try to eliminate the other uses of `from_inner_byte_pos`.